### PR TITLE
Fix Block and File backend registration

### LINKED ISSUE #1235: Cloud Block and File backend registration fails
**Issue/Feature Description:**
Cloud Block and File backend registration fails with backend check failure

**Why this issue to fixed / feature is needed(give scenarios or use cases):**
Blocking for File and Block service functionality

**How to reproduce, in case of a bug:**
Register backend by choosing block/file. It will fail

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by kumarashit**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind new feature
/kind bug fix
> /kind cleanup
> /kind revert change
> /kind design
> /kind documentation
> /kind enhancement

**What this PR does / why we need it**:
This PR fixes the issue of backend registration for File and Block. This is must fix
The BackendCheck() function is very specific to S3. So this should be called only for object storage type.
Added a check there.

**Which issue(s) this PR fixes**:

Fixes #1235 

**Test Report Added?**:
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
/kind TESTED
> /kind NOT-TESTED

**Test Report**:
image
image
image

Apart from this tested:

- [x] Azure Blob Registration
- [x] AWS S3 Registartion
- [x] Cloud bucket create 
- [x] Object Upload

**Special notes for your reviewer**:

### CODE REVIEW

File: api/pkg/backend/service.go [hunk 1]
```diff
@@ -47,6 +48,9 @@ const (
 	dataflowService_K8S    = "soda.multicloud.v1.dataflow"
 )
 
+// List of object storge providers supported by s3 services
+var object_types = []string{"aws-s3", "azure-blob", "ibm-cos", "hw-obs", "ceph-s3", "gcp-s3", "fusionstorage-object", "yig", "alibaba-oss", "sony-oda"}
```
Comment (AmitRoushan) [inline api/pkg/backend/service.go:52]: var objectStoarge = map[string]struct{}{
                                   "aws-s3": struct{}{},
                                    "azure-blob": struct{}{},
                                    "ibm-cos": struct{}{}, 
                                    "hw-obs": struct{}{},
                                     "ceph-s3": struct{}{},
                               }
func isObjectStorage(storage string) bool {
     _, ok := objectStoarge[storage ]
     return ok
}

IMHO, Above might be better implementation

**Review by AmitRoushan — CHANGES_REQUESTED** [review]

File: api/pkg/backend/service.go [hunk 2]
```diff
@@ -47,6 +48,9 @@ const (
 	dataflowService_K8S    = "soda.multicloud.v1.dataflow"
 )
 
+// List of object storge providers supported by s3 services
```
Comment (anvithks) [inline api/pkg/backend/service.go:51]: nit: object storge - > object storage

File: api/pkg/utils/utils.go [hunk 3]
```diff
@@ -215,3 +215,13 @@ func GetTimeoutSec(objSize int64) int64 {
 
 	return tmoutSec
 }
+
+// Function to check if the slice contains a string
+func SliceContainsString(inp string, list []string) bool {
```
Comment (PravinRanjan10) [inline api/pkg/utils/utils.go:220]: May be function can be renamed. ex: ObjectTyeStorageCheck(), this will give fair idea, during readability of code

**Comment by PravinRanjan10** [comment]
LGTM

File: api/pkg/backend/service.go (hunk 1 above)
Comment (kumarashit) [reply api/pkg/backend/service.go]: Thanks for the suggestion. Yes optimized it now. o(n) -> o(1) :)

File: api/pkg/backend/service.go [hunk 4]
```diff
@@ -246,12 +250,15 @@ func (s *APIService) CreateBackend(request *restful.Request, response *restful.R
...
-	_, err = s.s3Client.BackendCheck(ctx, backendDetailS3)
-	err1 := errors.New("Failed to register backend due to invalid credentials.")
-	if err != nil {
-		log.Errorf("failed to create backend due to wrong credentials: %v", err)
-		response.WriteError(http.StatusBadRequest, err1)
-		return
+    // This backend check will be called only for object storage
+	if utils.SliceContainsString(backendDetail.Type, object_types) {
+		_, err = s.s3Client.BackendCheck(ctx, backendDetailS3)
+		err1 := errors.New("Failed to register backend due to invalid credentials.")
```
[the remark this answers is not in the document]
Comment (kumarashit) [reply api/pkg/backend/service.go]: Moved it inside the loop

File: api/pkg/backend/service.go (hunk 2 above)
Comment (kumarashit) [reply api/pkg/backend/service.go]: done

**Review by AmitRoushan — APPROVED** [review]
LGTM

**Review by NajmudheenCT — APPROVED** [review]
LGTM

**Review by skdwriting — APPROVED** [review]
LGTM

**Pull request merged**